ssl/ssl_local.h

```diff
@@ -737,9 +744,25 @@ typedef struct ssl_ctx_ext_secure_st {
     unsigned char tick_aes_key[TLSEXT_TICK_KEY_LENGTH];
 } SSL_CTX_EXT_SECURE;
 
+/* Added from ciphersuite interface. */
+#define CIPHER_FLAG_FROM_CIPHERRSUITE   1
+/* Prioritize cipher on server when client does. */
+#define CIPHER_FLAG_PREFER              2
+/*
+ * Cipher belongs to group and it's not the last element, e.g.:
+ * cipherlist        [A:B:C]:D:[E]:[F:G]
+ * CIPHER_IN_GROUP    1 1 0  0  0   1 0
+ */
+#define CIPHER_IN_GROUP                 4
+
+struct ssl_cipher_list_flags_st {
+    STACK_OF(SSL_CIPHER) *cipher_list;
+    uint8_t *flags;
+};
+
 struct ssl_ctx_st {
     const SSL_METHOD *method;
-    STACK_OF(SSL_CIPHER) *cipher_list;
+    SSL_CIPHER_FLAGS *cipher_list_flags;
```

Comment:
I think it's kind of a shame if we have to make the new thing have a long name like this.  Changing the name in some way is probably a good idea, to make breakage more obvious, but perhaps something like ```ciphf_list``` could work?

Reply:
Ok, it's good idea. Done